src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketWebhookClient.java

```diff
@@ -3,19 +3,22 @@
 import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPage;
 import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhook;
 
+import java.util.stream.Stream;
+
 /**
  * A client to query for and register web hooks in Bitbucket Server.
  */
 public interface BitbucketWebhookClient {
 
     /**
-     * Fetch existing webhooks. Result could be further filtered by passing in event id filters.
+     * Returns a stream of existing webhooks. Result could be further filtered by passing in event id filters.
+     * every subsequent fetch of {@link BitbucketPage} results in a remote call to Bitbucket server.
      *
      * @param eventIdFilter, Event id filters. These ids are the same as the one recieved as
      *                       {@link com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhookSupportedEvents}
-     * @return a page of webhooks.
+     * @return a stream of pages of webhooks.
      */
-    BitbucketPage<BitbucketWebhook> getWebhooks(String... eventIdFilter);
+    Stream<BitbucketPage<BitbucketWebhook>> getWebhooks(String... eventIdFilter);
```

Comment:
I am not sure this is best way of doing `Stream`'s. I think what you really want to return is a `Stream` of `BitbucketWebhook`. I don't see any advantage in returning a `Stream` of `BitbucketPage`.
```
Stream<BitbucketWebhook> getWebhooks(String... eventIdFilter);
```

Reply:
I have fixed this.